### ISSUE TITLE
api: PutObjectStreaming fails with S3 for 0byte object. 

### ISSUE THREAD

**Issue by harshavardhana** [minio/minio-go#658]
```
go test -run TestPutObjectStreaming
--- FAIL: TestPutObjectStreaming (1.16s)
	api_functional_v4_test.go:398: Test 1 Error: The request signature we calculated does not match the signature you provided. Check your key and signing method. minio-go-test6jcyxtij4wwz0b6g3 test-object
FAIL
exit status 1
FAIL	github.com/minio/minio-go	1.162s
```

**Comment by harshavardhana** [comment]
Since I sent the fix already can you assign this to me? @deekoder

**Comment by harshavardhana** [comment]
Fixed by #657
